src/heliaServer.ts

```diff
@@ -1,23 +1,36 @@
-import { type FastifyReply, type FastifyRequest } from 'fastify'
+import { type FastifyReply, type FastifyRequest, type RouteGenericInterface } from 'fastify'
 import { CID } from 'multiformats'
+import { USE_SUBDOMAINS } from './constants.js'
 import { DEFAULT_MIME_TYPE, parseContentType } from './contentType.js'
 import { getCustomHelia } from './getCustomHelia.js'
 import { HeliaFetch } from './heliaFetch.js'
 import type debug from 'debug'
 
 const HELIA_RELEASE_INFO_API = (version: string): string => `https://api.github.com/repos/ipfs/helia/git/ref/tags/helia-v${version}`
 
-export interface RouteEntry {
+export interface RouteEntry<T extends RouteGenericInterface = RouteGenericInterface> {
   path: string
   type: 'GET' | 'POST'
-  handler: (request: FastifyRequest, reply: FastifyReply) => Promise<void>
+  handler: (request: FastifyRequest<T>, reply: FastifyReply) => Promise<void>
 }
 
-interface RouteHandler {
-  request: FastifyRequest
+interface RouteHandler<T extends RouteGenericInterface = RouteGenericInterface> {
+  request: FastifyRequest<T>
   reply: FastifyReply
 }
```

Comment:
i started to update all types but didn't want to get too lost. can revert this if desired.